src/pybind/mgr/stats/fs/perf_stats.py

```diff
@@ -233,15 +233,19 @@ def update_client_meta(self, rank_set):
         new_updates = {}
         pending_updates = [v[0] for v in self.client_metadata['in_progress'].values()]
         with self.meta_lock:
-            for rank in rank_set:
-                if rank in pending_updates:
-                    continue
-                tag = str(uuid.uuid4())
-                result = CommandResult(tag)
-                new_updates[tag] = (rank, result)
+            fsmap = self.module.get('fs_map')
+            for fs in fsmap['filesystems']:
+                mdsmap = fs['mdsmap']
+                for rank in rank_set:
+                    gid = mdsmap['up'][f"mds_{rank}"]
```

Comment:
I think we can just query rank0 gid for each file system. This would make things a lot easier (there is not need to gather the rank from incoming metrics in `get_raw_perf_counters`.

Reply:
In that case, there is `FSPerfStats.get_rank0_mds_gid_state` that returns rank0 mds's gid.

Reply:
@joscollin Changes include passing of fs name in **new_updates** dictionary,its easier to get rank0 gid here itself.